### DISCUSSION TITLE
Updating a file without using git at all

### DISCUSSION THREAD

**Discussion by jidanni** [community/community#23583] [New to GitHub]
<p>Sure I could do it the proper way:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://www.tutsmake.com/upload-project-files-on-github-using-command-line/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/1230959/181078030-b5713d65-fbe6-49c1-8218-ef0aab1e512d.png" class="site-icon" width="32" height="32">

      <a href="https://www.tutsmake.com/upload-project-files-on-github-using-command-line/" target="_blank" rel="noopener nofollow ugc" title="10:09AM - 29 November 2021">Tuts Make – 29 Nov 21</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/1230959/181078031-271bfbd5-84c5-49de-9fe7-09ad98d64306.jpeg" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd953417d61060d43d3a56e8fa34a4364977fe13_2_690x345.jpeg, https://aws1.discourse-cdn.com/github/original/3X/b/d/bd953417d61060d43d3a56e8fa34a4364977fe13.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/d/bd953417d61060d43d3a56e8fa34a4364977fe13.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd953417d61060d43d3a56e8fa34a4364977fe13_2_10x10.png"></div>

<h3><a href="https://www.tutsmake.com/upload-project-files-on-github-using-command-line/" target="_blank" rel="noopener nofollow ugc">Upload Project/Files On Github Using Command line - Tuts Make</a></h3>

  <p>Upload file project on github using command - today we would love to share with you how to upload the project and file using command line (cmd).</p>

  <p>
    <span class="label1">Est. reading time: 4 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Or the dumb way:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/jidanni/submission-cleaner/edit/main/riskfix">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/jidanni/submission-cleaner/edit/main/riskfix" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/1230959/181078032-6958a8c1-d2d5-4cdd-9313-e0e4bee60896.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/jidanni/submission-cleaner/edit/main/riskfix" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 73 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
and copy and paste my new content there.</p>
<p>But is there a way I can just do<br>
<code>$ scp riskfix github.com:jidanni/submission-cleaner/blob/main/riskfix</code><br>
I.e., just updating a file with one single shell command, not needing to<br>
use git at all?</p>

**Comment by jsoref** [comment]
<p>If you’re uninterested in learning <code>git</code>, in theory you could try using <code>svn</code> instead:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/github/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/github/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181078036-bf94d664-3113-4a07-b74c-bb76d7b27eac.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/github/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients" target="_blank" rel="noopener nofollow ugc">Support for Subversion clients - GitHub Docs</a></h3>

  <p>GitHub repositories can be accessed from both Git and Subversion (SVN) clients. This article covers using a Subversion client on GitHub and some common problems that you might run into.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>You could of course create a cli that wraps the <code>edit</code> entrypoint. But note that technically that <code>edit</code> entrypoint basically creates a file change and then creates a commit:</p>
<p>I think it’s effectively something like this:</p><aside class="onebox githubgist" data-onebox-src="https://gist.github.com/testcollab/1236348/8ebe7df22fef7e695b2835614aaeea0114a3477d">
  <header class="source">

      <a href="https://gist.github.com/testcollab/1236348/8ebe7df22fef7e695b2835614aaeea0114a3477d" target="_blank" rel="noopener nofollow ugc">gist.github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://gist.github.com/testcollab/1236348/8ebe7df22fef7e695b2835614aaeea0114a3477d" target="_blank" rel="noopener nofollow ugc">https://gist.github.com/testcollab/1236348/8ebe7df22fef7e695b2835614aaeea0114a3477d</a></h4>

  <h5>an_example_usage.rb</h5>
  <pre><code class="Ruby">require 'editor.rb'

ed = Editor.new('testcollab/rails-test')
if ed.update_file('README.txt', 'my message', 'my new content')
  ed.set_author('Scott', 'schacon@gmail.com')
  ed.update_file('README.2.txt', 'my message', 'my new content')
end</code></pre>
  <h5>editor.rb</h5>
  <pre><code class="Ruby">require 'rubygems'
require 'httparty'
require 'json'

class Editor
  include HTTParty

  attr_accessor :repo, :branch, :author

  base_uri 'https://api.github.com'</code></pre>
    This file has been truncated. <a href="https://gist.github.com/testcollab/1236348/8ebe7df22fef7e695b2835614aaeea0114a3477d" target="_blank" rel="noopener nofollow ugc">show original</a>

<p>
</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Note: you can’t use passwords for this api, you could use a PAT…</p>

**Comment by jfullstackdev** [comment]
<p>you can use GitHub Web to upload, edit files, without Git at all and any other command line</p>
<p>to upload, simply drag and drop on the web,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/8/b88fbb50b28fb6775e1d67709bdbebaaccbe448f.png" data-download-href="/uploads/short-url/qkHNuHAJlKk3Jq9I3DB6MSqvAS3.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181078040-a2d93191-d8c6-4a92-88a3-44e796db0b46.png" alt="image" data-base62-sha1="qkHNuHAJlKk3Jq9I3DB6MSqvAS3" width="690" height="410" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b88fbb50b28fb6775e1d67709bdbebaaccbe448f_2_690x410.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b88fbb50b28fb6775e1d67709bdbebaaccbe448f_2_1035x615.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/8/b88fbb50b28fb6775e1d67709bdbebaaccbe448f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/8/b88fbb50b28fb6775e1d67709bdbebaaccbe448f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1186×706 20 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>to edit,</p>
<p>click a certain file and look for the edit button,</p>
<p><img src="https://user-images.githubusercontent.com/47092464/181078041-c26c1bd0-5c04-453c-995c-333a8a1b2df1.png" alt="image" data-base62-sha1="s2cNEnBopU3GZqx75Y79c1aYwb8" width="491" height="208"></p>

**Comment by byrneh** [comment, marked as answer]
<p>The Contents API endpoints let you create, modify, and delete Base64 encoded content in a repository. To request the raw format or rendered HTML (when supported), use custom media types for repository contents</p>
<p><a href="https://docs.github.com/en/rest/reference/repos#contents" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/repos#contents</a></p>

**Comment by jidanni** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/1230959/181078045-f9a264cc-f448-4c30-a807-092610c34dc5.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents" target="_blank" rel="noopener nofollow ugc">Repositories - GitHub Docs</a></h3>

  <p>The Repos API allows to create, manage and control the workflow of public and private GitHub repositories.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
